Warehouse.Client/warehouse-client/src/app/customer-form/customer-form.component.ts

```diff
@@ -0,0 +1,34 @@
+import {Component} from '@angular/core';
+import {FormControl, FormGroup, Validators} from "@angular/forms";
+import {MyErrorStateMatcher} from "../errors/myErrorStateMatcher";
+import {Customer} from "../models/customer" ;
+import {MAT_DIALOG_DATA, MatDialogRef} from "@angular/material/dialog";
+
+@Component({
+  selector: 'app-customer-form',
+  templateUrl: './customer-form.component.html',
+  styleUrls: ['./customer-form.component.css']
+})
+export class CustomerFormComponent {
+  firstName!: string;
+  lastName!: string;
+  formControl = new FormGroup({
+    firstName: new FormControl('', [Validators.required, Validators.minLength(5)]),
+    lastName: new FormControl('', Validators.minLength(5)),
+    email: new FormControl('', [Validators.required, Validators.email]),
+    phone: new FormControl('', [Validators.required, Validators.pattern('^\\(?([0-9]{3})\\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$')])
```

Comment:
You can move such patterns to some constant class instead of using a string literal